test/unit/org/apache/cassandra/service/opstate/CleanupStateTrackerTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.opstate;
+
+
+import java.io.File;
+import java.time.Instant;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class CleanupStateTrackerTest
+{
+
+    @After
+    public void afterEach(){
+        new File(OpStateTestConstants.TEST_STATE_FILE_LOCATION).delete();
```

Comment:
nit: you can use `Files.createTempFile()` which should do some clean up for you

Reply:
That creates the file in a temp directory correct? Problem is the StatePersister uses the normal `new File` method. I could probably provide a tmp directory path instead of the current `TEST_STATE_FILE_LOCATION` but then I am unclear about when the File would  actually get deleted and if different Test would erroneously interact with the same File.

Reply:
I suppose you could do something like:
```
    private static Path directory;
    private static Path path;
    
    @BeforeClass
    public static void before() throws IOException
    {
         directory = Files.createTempDirectory("test-dir");
         path = directory.resolve("actual-file");
    }

    @After
    public void afterEach()
    {
        new File(path.toString()).delete();
    }
```
Which would provide the actual file path like you have above (side note can the KeyspaceTableOpStatePersister take a `Path` instead of a `String` in the constructor?)